File: bending-api/src/main/java/pw/dotdash/bending/api/bender/BenderService.java
package pw.dotdash.bending.api.bender;

import org.spongepowered.api.Sponge;
import org.spongepowered.api.entity.living.player.Player;

import java.util.Optional;
import java.util.UUID;

/**
 * Represents a service for managing benders.
 */
public interface BenderService {

    /**
     * Gets the singleton instance of the {@link BenderService}.
     *
     * @return The singleton service instance
     */
    static BenderService getInstance() {
        return Sponge.getServiceManager().provideUnchecked(BenderService.class);
    }

    /**
     * Gets a {@link Bender} player by their UUID.
     *
     * @param uniqueId The UUID to get the bender player from
     * @return The bender, if available
     */
    Optional<Bender> getBender(UUID uniqueId);

    /**
     * Gets a {@link Bender} player from the given {@link Player}.
     *
     * <p>If a bender does not already exist with the specified {@link UUID},
     * it will be created.</p>
     *
     * @param player The player to get a bender for
     * @return The bender
     */
    Bender getOrCreateBender(Player player);
}

File: src/main/kotlin/arvenwood/bending/api/service/BenderService.kt
package arvenwood.bending.api.service

import arvenwood.bending.api.Bender
import org.spongepowered.api.Sponge
import java.util.*

interface BenderService {

    companion object {
        @JvmStatic
        fun get(): BenderService =
            Sponge.getServiceManager().provideUnchecked(BenderService::class.java)
    }

    /**
     * Gets a [Bender] by their [UUID].
     *
     * @param uniqueId The UUID to get the bender from
     * @return The [Bender]
     */
    operator fun get(uniqueId: UUID): Bender
}